src/python/pants/backend/python/util_rules/pex_cli.py

```diff
@@ -80,21 +81,30 @@ async def setup_pex_cli_process(
     pex_env: PexEnvironment,
     python_native_code: PythonNativeCode,
 ) -> Process:
-    downloaded_pex_bin = await Get(
-        DownloadedExternalTool, ExternalToolRequest, pex_binary.get_request(Platform.current)
+    tmpdir = ".tmp"
```

Comment:
Oh, this took me a minute. The reason this works with `--local-execution-root-dir` is that this is a relative path. Consider adding a comment like:

> NB: this is a relative path so that it will be relative to where the Process runs, which is controlled by --local-execution-root-dir.

Reply:
You may have missed all the local vs remote kvetching in the big comment and the PR description / commit message. There is no `--remote-execution-root-dir` - that's up to the remoting implementation, and we must work under all those too - so this _has_ to be relative.